File: SolucaoSistemaPenal/API/Models/Detento.cs
namespace API.Models;

public class Detento : Pessoa
{
  public string DetentoId { get; set; } = Guid.NewGuid().ToString();
  // public int TempoPenaInicial { get; set; }
  // public int PenaRestante { get; set; }
  public DateTime InicioPena { get; set; }
  public DateTime FimPena { get; set; }
  public List<Atividade> Atividades { get; set; } = [];
}

File: front/src/interfaces/Detento.ts
import { Atividade } from "./Atividade";
import { Pessoa } from "./Pessoa";

export interface Detento extends Pessoa {
    detentoId? : string;
    tempoPenaInicial : number;
    penaRestante : number;
    inicioPena : string;
    fimPena : string;
    atividades: Atividade[];
}